### PR TITLE
feat: single line navigation

### PR DESCRIPTION
**Pull request by Lelith**
# Purpose of PR
reduce the size and structure of the Navbar to a single-line.
It contains the following breaking changes:

- removal of Components `NavbarSearch` and `NavbarHelp`
- Prop change of `NavbarItem`: 
  - `title` is no longer required
  -  accepts new property `label` which will be set as `aria-label` for Items without a title 
- Prop changes of `Navbar`
  - removes the following props:   
    - `badge`
    - `bottomRightItems`
    - `children`
    - `help`
    - `search`
    - `topRightItems`
  - introduces new props
    - `mainNav` as replacement for `children`
    -  `secondaryNav` as a replacement for the other removed properties   
    
Screenshots:
     
image

image

### CODE REVIEW

File: packages/components/navbar/src/Navbar.tsx [hunk 1]
```diff
@@ -9,33 +9,33 @@ import { getNavbarStyles } from './Navbar.styles';
...
    */
-  bottomRightItems?: React.ReactNode;
-  children?: React.ReactNode;
+  logo?: React.ReactNode;
+
+  /** Environment Switcher component */
+  switcher?: React.ReactNode;
+
+  /** Main Navigation Elements */
+  mainNav?: React.ReactNode;
```
Comment (denkristoffer) [inline packages/components/navbar/src/Navbar.tsx:22]: Can you write out these names?

File: package-lock.json
Comment (cf-remylenoir) [inline package-lock.json:1]: Is it intentional? There aren't changes in the `package.json`.

Comment (Lelith) [reply package-lock.json:1]: hm, strange. i did a rebase and needed to npm install again i did not apply something like that on purpose, will revert.

File: packages/components/navbar/src/NavbarItemIcon/NavbarItemIcon.styles.ts [hunk 2]
```diff
@@ -6,6 +6,9 @@ export const getNavbarItemIconStyles = () => ({
     '&:last-child': {
       marginLeft: tokens.spacing2Xs,
     },
+    '&:only-child': {
```
Comment (cf-remylenoir) [inline packages/components/navbar/src/NavbarItemIcon/NavbarItemIcon.styles.ts:9]: TIL about the `only-child` pseudo-class 😄

File: packages/components/navbar/src/CompoundNavbar.ts
Comment (cf-remylenoir) [inline packages/components/navbar/src/CompoundNavbar.ts:1]: I wonder what is the boundary between `Navbar.TopbarItem` and `Navbar.Item` now 🤔  Should we consolidate everything into `Navbar.Item`?

**Comment by cf-remylenoir** [comment]
Not related to your changes but I noticed we have an a11y issue with the account button.

Screenshot 2024-06-10 at 14 21 34

**Comment by damann** [comment]
Does that mean the Avatar component has the same issues? Or is it only in the navigation?
> Not related to your changes but I noticed we have an a11y issue with the account button.

**Comment by cf-remylenoir** [comment]
> Does that mean the Avatar component has the same issues? Or is it only in the navigation?
> 
> > Not related to your changes but I noticed we have an a11y issue with the account button.



> Does that mean the Avatar component has the same issues? Or is it only in the navigation?
> 
> > Not related to your changes but I noticed we have an a11y issue with the account button.

Only in the navigation where we wrap the avatar with a button to display the account settings.

File: packages/components/navbar/src/CompoundNavbar.ts
Comment (Lelith) [reply packages/components/navbar/src/CompoundNavbar.ts:1]: I think this component can be deleted like the Help or Search Helper Components.

**Review by denkristoffer — COMMENTED** [review]
Can we make the root element `as="header"` for better accessibility?

File: packages/components/navbar/src/Navbar.tsx [hunk 3]
```diff
@@ -9,33 +9,33 @@ import { getNavbarStyles } from './Navbar.styles';
...
+  secondaryNavigation?: React.ReactNode;
+
+  /** User Account Component  */
+  account?: React.ReactNode;
+
   /**
    * Defines the max-width of the content inside the navbar.
    * @default '100%'
    */
   contentMaxWidth?: string;
```
[the remark this answers is not in the document]
Comment (Lelith) [reply packages/components/navbar/src/Navbar.tsx:34]: So we would only have fullscreen and the wide variant with the fixed width? 
i think we can talk about this again on the responsive task

[the remark this answers is not in the document]
Comment (denkristoffer) [reply packages/components/navbar/src/Navbar.tsx:34]: Yeah basically controlling the width with the variant. But it makes sense to deal with later.

**Review by cf-remylenoir — APPROVED** [review]
Looking great! ✨

**Pull request merged**